### PR TITLE
Revert "Revert "Get end points""

### PR DESCRIPTION
**Pull request by RaphealOlumofe**
Reverts raphaelolu/starterProject#25

### CODE REVIEW

File: domain/build.gradle [hunk 1]
```diff
@@ -1,15 +1,12 @@
 
 dependencies {
     annotationProcessor 'org.projectlombok:lombok:1.18.12'
-
     compile group: 'com.fasterxml.jackson.core', name: 'jackson-core', version: '2.10.3'
     compile group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.10.3'
-
-
-
+    compile 'org.springframework.boot:spring-boot-starter-web:2.0.1.RELEASE'
```
Comment (petermcneil) [inline domain/build.gradle:6]: We don't want or need these dependencies in the domain sub project

File: domain/src/main/java/starter/project/domain/Car.java [hunk 2]
```diff
@@ -5,12 +5,18 @@
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
+import javax.persistence.Entity;
+import javax.persistence.Id;
```
Comment (petermcneil) [inline domain/src/main/java/starter/project/domain/Car.java]: we don't want to add these just yet

File: domain/src/main/java/starter/project/domain/Doors.java [hunk 3]
```diff
@@ -6,11 +6,14 @@
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
+import javax.persistence.Embeddable;
```
Comment (petermcneil) [inline domain/src/main/java/starter/project/domain/Doors.java]: or thks

File: domain/src/main/java/starter/project/domain/Engine.java [hunk 4]
```diff
@@ -6,11 +6,14 @@
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
+import javax.persistence.Embeddable;
```
Comment (petermcneil) [inline domain/src/main/java/starter/project/domain/Engine.java]: or this

File: domain/src/main/java/starter/project/domain/Transmission.java [hunk 5]
```diff
@@ -6,11 +6,14 @@
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
+import javax.persistence.Embeddable;
```
Comment (petermcneil) [inline domain/src/main/java/starter/project/domain/Transmission.java]: or this

File: domain/src/main/java/starter/project/domain/Wheels.java [hunk 6]
```diff
@@ -4,11 +4,15 @@
 import lombok.Builder;
 import lombok.Data;
 import lombok.NoArgsConstructor;
+
+import javax.persistence.Embeddable;
```
Comment (petermcneil) [inline domain/src/main/java/starter/project/domain/Wheels.java]: or this

File: frontend/build.gradle [hunk 7]
```diff
@@ -2,26 +2,21 @@ plugins {
...
-
     testCompile group: 'junit', name: 'junit', version: '4.12'
-    implementation 'org.springframework.boot:spring-boot-starter-web'
     compile project(':domain')
     testImplementation('org.springframework.boot:spring-boot-starter-test') {
         exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
         implementation 'com.google.code.gson:gson:2.8.6'
-
+        compile 'org.springframework.boot:spring-boot-starter-web:2.0.1.RELEASE'
+       compile group: 'com.h2database', name: 'h2', version: '1.3.148'
```
Comment (petermcneil) [inline frontend/build.gradle:18]: don't want any database stuff in this commit

File: frontend/src/main/java/olumofe/raphael/frontend/Application.java [hunk 8]
```diff
@@ -1,7 +1,8 @@
-package starter.project.frontend;
+package olumofe.raphael.frontend;
```
Comment (petermcneil) [inline frontend/src/main/java/olumofe/raphael/frontend/Application.java]: package name has changed here. Could you change that across the project? Or revert it here

File: frontend/src/main/java/olumofe/raphael/frontend/CarController.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+            }
+        }
+        return null;
+    }
+
+    @GetMapping(path = "/cars/{list}")
+    public List<Car> getBatchCars(@PathVariable List list) {
+
+        List<Car> newList = new ArrayList<>();
+        for (Map.Entry<String, Car> m : map.entrySet()) {
```
Comment (petermcneil) [inline frontend/src/main/java/olumofe/raphael/frontend/CarController.java]: I am unsure what this Map is doing, can you please explain?

File: frontend/src/main/java/olumofe/raphael/frontend/StatusController.java [hunk 10]
```diff
@@ -1,10 +1,10 @@
-package starter.project.frontend;
+package olumofe.raphael.frontend;
```
Comment (petermcneil) [inline frontend/src/main/java/olumofe/raphael/frontend/StatusController.java]: package name

File: frontend/src/main/java/olumofe/raphael/frontend/StatusController.java [hunk 11]
```diff
@@ -1,10 +1,10 @@
...
 import org.springframework.web.bind.annotation.*;
-import starterProject.domain.*;
 import java.util.ArrayList;
 import java.util.List;
 @org.springframework.stereotype.Controller
-public class Controller
+@RestController
```
Comment (petermcneil) [inline frontend/src/main/java/olumofe/raphael/frontend/StatusController.java]: Two controllers here, choose one

File: frontend/src/test/java/olumofe/raphael/frontend/CarControllerTest.java [hunk 12]
```diff
@@ -1,4 +1,4 @@
-package starter.project.frontend;
+package olumofe.raphael.frontend;
```
Comment (petermcneil) [inline frontend/src/test/java/olumofe/raphael/frontend/CarControllerTest.java]: package name, and removed some tests

**Review by petermcneil — CHANGES_REQUESTED** [review]

File: frontend/src/main/java/olumofe/raphael/frontend/CarController.java (hunk 9 above)
Comment (RaphealOlumofe) [reply frontend/src/main/java/olumofe/raphael/frontend/CarController.java]: its going through the map and if list contains a key in the map the value of that key being added to the list and returned

Comment (timlarochebbc) [reply frontend/src/main/java/olumofe/raphael/frontend/CarController.java]: https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#getOrDefault-java.lang.Object-V-

File: frontend/src/test/java/olumofe/raphael/frontend/CarControllerTest.java (hunk 12 above)
Comment (RaphealOlumofe) [reply frontend/src/test/java/olumofe/raphael/frontend/CarControllerTest.java]: resolved

File: frontend/src/main/java/olumofe/raphael/frontend/StatusController.java (hunk 10 above)
Comment (RaphealOlumofe) [reply frontend/src/main/java/olumofe/raphael/frontend/StatusController.java]: resolved

File: frontend/src/main/java/olumofe/raphael/frontend/Application.java (hunk 8 above)
Comment (RaphealOlumofe) [reply frontend/src/main/java/olumofe/raphael/frontend/Application.java]: resolved

**Pull request closed without merging**